### ISSUE TITLE
Authentication via token, how I can get the token key?

### ISSUE THREAD

**Issue by anhskohbo** [GrahamCampbell/Laravel-Bitbucket#4]
Hi, thank for great project, I'm working with bitbucket and this is very helpful for me.

First, I'm try authentication via basic, everything is OK. But switch to token method, I can't find where is token code (I have try **App passwords** code, that's similar **Personal access tokens** in Github but is doesn't work).

I found this doc: https://gentlero.bitbucket.io/bitbucket-api/0.8/examples/authentication.html#oauth2-authorization-code-3-legged-flow, It say require user authentication via browser to get `access_token`. That seem not right way here.

For now, I write own a class `OAuthAuthenticator` use `client_id`, `client_secret` to authentication. It's worked for me.

And seem it's good method. Maybe need a pull-request for that?

**Comment by GrahamCampbell** [comment]
We now support oauth.

**Comment by jee7** [comment]
Had the same issue. I could not find a way to generate a permanent access token from BitBucket either. All I found was that they use OAuth 2, which only generates temporary (couple of hours) tokens and a longer refresh token. The current implementation here in Laravel-Bitbucket does not seem to support that - fetching and refreshing the tokens using the OAuth `key` and `secret` that BitBucket web interface allows to generate.
https://confluence.atlassian.com/bitbucket/oauth-on-bitbucket-cloud-238027431.html

I did find one doc page, which talked about personal access tokens, but that seems to be for when you set up your own BitBucket Server. Could not find it in the BitBucket.org web interface.
https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html

So the solution was to use the App passwords. It seems to work if you use the authentication method `password` and the username is your own username. I guess the idea is that you won't put your personal password in the config file, but the generated one.
